File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/service/analytics/MembershipVideoDownloadedAction.java
package com.worldventures.dreamtrips.social.ui.membership.service.analytics;

import com.worldventures.core.service.analytics.AdobeTracker;
import com.worldventures.core.service.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.BaseAnalyticsAction;

@AnalyticsEvent(action = "membership",
                trackers = {AdobeTracker.TRACKER_KEY})
public class MembershipVideoDownloadedAction extends BaseAnalyticsAction {

   @Attribute("video_id")
   String videoName;

   @Attribute("download") final String download = "1";

   public MembershipVideoDownloadedAction(String videoName) {
      this.videoName = videoName;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/ui/membership/service/analytics/MembershipVideoDownloadedAction.kt
package com.worldventures.dreamtrips.social.ui.membership.service.analytics

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.core.service.analytics.BaseAnalyticsAction
import com.worldventures.janet.analytics.AnalyticsEvent

@AnalyticsEvent(action = "membership", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class MembershipVideoDownloadedAction(@field:Attribute("video_id")
                                      internal var videoName: String) : BaseAnalyticsAction() {

   @field:Attribute("download") internal val download = "1"
}
